### DISCUSSION TITLE
Command to convert .jpg to .avif?

### DISCUSSION THREAD

**Discussion by eugene-su** [ImageMagick/ImageMagick#2388] [Help] [answered]
Hi,

AFAIK, ImageMagick support `AVIF* HEIC rw+ AV1 Image File Format (1.7.0)`. But I can't find proper command line for converting .jpg to .avif.

Thank you.

**Comment by snibgo** [comment]
What version of IM, on what platform?

Try:
```
magick in1.jpg in2.jpg in3.jpg out.avif
```

**Reply by eugene-su** [reply]
Thank you for response.
I have
```
Version: ImageMagick 7.0.10-25 Q16 x86_64 2020-08-01 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo djvu fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png raqm raw rsvg tiff webp wmf x xml zlib
```
After typing `magick in1.jpg out.avif` i get `magick: Success 't.avif' @ error/heic.c/IsHeifSuccess/136.`

**Comment by eugene-su** [comment]
@snibgo, on x86_64 GNU/Linux.

**Comment by dooman87** [comment]
I'm having the same here on Debian 10

**Comment by eugene-su** [comment, marked as answer]
The answer on the question here: https://github.com/ImageMagick/ImageMagick/issues/1432.

**Comment by rubo77** [comment]
This converts images losslessly to avif, but how do I turn on lossy? I would like to lower the quality to a minimum, that is still acceptable

**Reply by dooman87** [reply]
I think IM could only do lossy conversion to AVIF prior version 7.0.10-58. In 7.0.10-58 if you set quality to 100 then it will use lossless compression

**Reply by rubo77** [reply]
I think IM uses a very high quality as default and the `-quality` value has no effect on avif files. 

If I use [libavif](https://stackoverflow.com/a/65765495/1069083) instead, I get much smaller files

**Comment by azev** [comment]
`magick in1.jpg in2.jpg in3.jpg out.avif`
For 3x 1080p images I get **extremely high cpu and memory usage** and `out.avif` gets 0bytes.

i7 8Gb Ram. Windows 10 x64
Version: ImageMagick 7.0.10-60 Q8 x64 2021-01-25

**Comment by JasonNeuswanger** [comment]
I'm using 7.0.11-10 Q16 x86_64 2021-04-24 on CentOS and my .avif files are coming out huge, like 15X bigger than jpeg or webp in the same resolution for the same image.

**Reply by forke88** [reply]
Have the same problem with  ImageMagick 7.1.1-19 Q16-HDRI x86_64 21601 on Centos. Have you managed to solve this?

**Comment by zoobab** [comment]
I am looking to do the reverse AVIF2JPG, any idea?

**Reply by snibgo** [reply]
You can probably do that with ffmpeg.
